include/swift/AST/DiagnosticsSema.def

```diff
@@ -2518,6 +2518,11 @@ NOTE(decodable_suggest_overriding_init_here,none,
 NOTE(codable_suggest_overriding_init_here,none,
      "did you mean to override 'init(from:)' and 'encode(to:)'?", ())
 
+NOTE(missing_member_type_conformance_prevents_synthesis, none,
+     "%select{associated value|stored property}0 type %1 does not conform to "
```

Comment:
Could I make a suggestion?

Since the note pops up next to the type in question, I don’t think we get much mileage out of saying it’s a stored property or associated value, as this would be self-evident.

I do think it’s important to clarify what is prevented, since that’s not immediately obvious. In other words, prevents “automatic conformance synthesis” of what? (Also, synthesis as it’s used here implies automatic in the sense it’s used here.)

Perhaps, “type %1 does not conform to protocol %2, preventing synthesized conformance of %3 to %2”, where %3 is the containing type.

Reply:
@xwu you make some very good points, I've adopted most of your suggestions. I (weakly) disagree that the stored property/associated value labeling is unnecessary though, so I've left it in for now. To someone less familiar with the synthesized conformance rules, I think it's important to explicitly call out that an associated value or stored property type is blocking synthesis as opposed to a computed property type which would not.